### PR TITLE
[READY] Adds the Sleeper

### PR DESCRIPTION
**Pull request by EOBGames**
## About The Pull Request
The _cryosleeper_, that is! As requested by plapatin, this adds the Chill-out-Chamber, a cryo sleeper to give a degree of security when going SSD. Working (suspiciously) similarly to personal lockers, you can claim them by swiping your ID on them. From there, simply hop in, lock it up, and enjoy your cryosleep! Inside, you'll be relatively secure, with the pod being unweldable and starting bolted. Cracking open a sleeper is the same as opening a locker- use an emag to be quick, or break it open with your gamer gear of choice. Additionally, while the pod is not occupied, you can remove the ID lock using a multitool.

Also included is some feedback for actions performed on personal closets.

Big thanks to @Couls for unfucking the new multitool code.

The sleepers can be found in the dorms on every map, the mining base, and on Delta, in each head's personal quarters.

## Why It's Good For The Game
A lot of people have been requesting a secure method of going SSD, particularly on Manuel, and I feel this satisfies that demand without making the occupant completely invincible to outside forces.

## Changelog
:cl:
add: Nanotrasen has sent the latest in cryogenic technology to the station: the Chill-out-Chamber, perfect for going SSD!
/:cl:

### CODE REVIEW

File: code/game/objects/structures/crates_lockers/closets/secure/personal.dm [hunk 1]
```diff
@@ -35,6 +36,23 @@
 	new /obj/item/instrument/piano_synth(src)
 	new /obj/item/radio/headset( src )
 
+// Plapatin owes me 3 TF2 keys for this, don't let him forget it
```
Comment (Couls) [inline code/game/objects/structures/crates_lockers/closets/secure/personal.dm]: who wrote this comment

**Comment by cacogen** [comment]
seems familiar somehow

**Comment by Ohagi-Chan** [comment]
Are you safe from rad storms?

File: code/game/objects/structures/crates_lockers/closets/secure/personal.dm [hunk 2]
```diff
@@ -50,8 +68,21 @@
...
+				to_chat(user, "<span class='notice'>You swipe your ID and take ownership of the [src].</span>")
 		else
 			to_chat(user, "<span class='danger'>Access Denied.</span>")
 	else
 		return ..()
+
+// This will let you wipe the owner data, for business or pleasure (or just to reset it for the next user, you sick bastard)
+/obj/structure/closet/secure_closet/personal/sleeper/attackby(obj/item/W, mob/user, params)
+	if(istype(W, /obj/item/multitool))
+		if(registered_name != null)
```
Comment (PKPenguin321) [inline code/game/objects/structures/crates_lockers/closets/secure/personal.dm]:
```suggestion
		if(registered_name)

```

**Comment by itseasytosee** [comment]
I'm glad you didn't make them invincible SSD cages. Any old changling with a multitool can crack one of these bad boys open for a quick snack

File: code/game/objects/structures/crates_lockers/closets/secure/personal.dm [hunk 3]
```diff
@@ -50,8 +68,21 @@
...
-				desc = "Owned by [I.registered_name]."
+				desc += " Owned by [I.registered_name]."
+				to_chat(user, "<span class='notice'>You swipe your ID and take ownership of the [src].</span>")
 		else
 			to_chat(user, "<span class='danger'>Access Denied.</span>")
 	else
 		return ..()
+
+// This will let you wipe the owner data, for business or pleasure (or just to reset it for the next user, you sick bastard)
+/obj/structure/closet/secure_closet/personal/sleeper/attackby(obj/item/W, mob/user, params)
```
Comment (ATH1909) [inline code/game/objects/structures/crates_lockers/closets/secure/personal.dm]: make this also happen if the pod is emagged

Comment (ATH1909) [reply code/game/objects/structures/crates_lockers/closets/secure/personal.dm]: or use another emagging effect, idk

**Comment by Shadowflame909** [comment]
I'm a little confused.

Isn't it too weak?

Wouldn't it be safer to hide in a locker at lavaland, or sleep in dorms?

Whats the upside to this here. Dorms has door bolts as well.

**Comment by Grazyn** [comment]
Few things attract greyshirts more than a bolted dorm. Even if they don't loot you, they will at the very least break down the wall out of curiosity, ruining your seclusion anyway.

The idea is good, I just hope it doesn't result in people who SSD in the open becoming free game because "they would've used a sleeper if they wanted to be left alone"

**Comment by MMMiracles** [comment]
There really should be some sort of alert for when these sleepers are popped open not by the occupant, otherwise I'd still rather hide in a locker or unpowered disposal bin. Maybe ping the medical/security channel when one is forced open?

**Comment by MishaTarkus** [comment]
This brings back a constant suggestion from a few years back where we use these as the standard "enter shift" mechanism rather than the arrivals shuttle. Always bothered me that we apparently have a shuttle that can head from outside space but not help leave it, and also the implication RP wise that this is the first time everyone is meeting up (or at least, everyone arriving by arrivals)

Cryosleepers in "spawn" could be invincible and stabilize people until they step out. So similar to the Arrivals shuttle there's a safety net before you head out into a destroyed Arrivals.

**Comment by HomocidlePancakes** [comment]
My fucking god if I see somebody else use SSD inappropriately I am going to pop a vein. SSD stands for Sudden Sleep Disorder. SUDDEN. “Hey guys, I’ll go SSD for 5 minutes”, fucking shut the fuck and eat a dick. You can’t intentionally go SSD. For the love of god, cease and fucking desist.

**Comment by MMMiracles** [comment]
you're about to go SSD forever if you keep that up

**Comment by carlarctg** [comment]
> My fucking god if I see somebody else use SSD inappropriately I am going to pop a vein. SSD stands for Sudden Sleep Disorder. SUDDEN. “Hey guys, I’ll go SSD for 5 minutes”, fucking shut the fuck and eat a dick. You can’t intentionally go SSD. For the love of god, cease and fucking desist.

you're being very OCD disorder with this

**Comment by blessedmulligan** [comment]
> 
> 
> My fucking god if I see somebody else use SSD inappropriately I am going to pop a vein. SSD stands for Sudden Sleep Disorder. SUDDEN. “Hey guys, I’ll go SSD for 5 minutes”, fucking shut the fuck and eat a dick. You can’t intentionally go SSD. For the love of god, cease and fucking desist.

"Sudden" doesn't mean "instantaneous." Besides, SSD can also stand for "space sleep disorder." SSD is just a convenient way to explain going afk IC. If you're security or a head of staff or a member of a teamwork-dependent department, it's important that people know you're AFK and not disconnected or dead. If you say "nuuuuuu you can't say you're going SSD my RP!" people will just come up with something else:
"Hey guys I'm really stressed, gonna do some deep meditation for a few minutes."
"Oh wow, it's already time to pray to Ratvar, which involves standing perfectly still and not responding to my surroundings for an indeterminate amount of time"

just let people use SSD because it's a part of server culture. Pretend it gives you migraine auras before an attack or something.

**Comment by SkeletalElite** [comment]
> Few things attract greyshirts more than a bolted dorm. Even if they don't loot you, they will at the very least break down the wall out of curiosity, ruining your seclusion anyway.
> 
> The idea is good, I just hope it doesn't result in people who SSD in the open becoming free game because "they would've used a sleeper if they wanted to be left alone"

I think you're being a little optimistic in your hope that this will somehow ward off grey shirts from looting you. Any grey shirt with a multitool sees one of these in use who wants some free loot will claim it. I think shadowflame is right when he says it would be safer to hide in a locker somewhere.

File: code/game/objects/structures/crates_lockers/closets/secure/personal.dm (hunk 3 above)
Comment (EOBGames) [reply code/game/objects/structures/crates_lockers/closets/secure/personal.dm]: keep in mind these are instanced from lockers- emagging should open and break them, which is functionally identical to removing the owner data

**Comment by MishaTarkus** [comment]
> > snippety
> snip

I'm a big fan of "I'm taking a smoke break" when on Head roles, just light a big fat cigar and buckle myself to my chair and pretend to be a fucking corpse

Seriously though I actually sorta agree in that SSD is better as an IC justification for sudden AFK and Sleepytimes/nap is better for planned ones, adds that immershun and allows the catpeople to try and ERP-jump you in bed

little do they know my pillow buddy's a shotgun, oo-rah

**Comment by AWalton** [comment]
Make them take two minutes to open with a multitool. Not impenetrable but someone has to be really committed if they want to check who's been in there for so long. 

As for location: next to recreation area add a room of maybe 6 of these.

**Comment by spessbandit** [comment]
why not just port this from someplace

**Comment by cacogen** [comment]
https://tgstation13.org/phpBB/viewtopic.php?f=9&t=25258

**Comment by HomocidlePancakes** [comment]
>Wall of text by blessedmulligan 

Saying “I am going SSD” is the same as “I have epilepsy and I am going to give myself a seizure.” It’s a disorder ffs.

>part of the server culture

*WHEEZE*
No, it’s not. It’s a term that migrated from bay and it would be fine, if people knew what it actually meant. Every time I see it being used I cringe harder than after seeing a “lol” IC.

**Comment by cacogen** [comment]
> It’s a term that migrated from bay and it would be fine

It's never fine.

File: code/game/objects/structures/crates_lockers/closets/secure/personal.dm [hunk 4]
```diff
@@ -3,6 +3,7 @@
 	name = "personal closet"
 	req_access = list(ACCESS_ALL_PERSONAL_LOCKERS)
 	var/registered_name = null
+	var/original_desc = "It's a secure locker for personnel. The first card swiped gains control." // a hacky fix, to be sure, but it works.
```
Comment (XDTM) [inline code/game/objects/structures/crates_lockers/closets/secure/personal.dm]: you can use initial(desc) instead of doing this

**Comment by kopoba** [comment]
Okay russian server got cryosleepers. Let me explain mechanics.
You can lay there by yourself or you can put there sleeping man. If you goin there by yourself after ghosting you just vanish if its some other dude it take some time to suck him in long term storage like 10-15 minutes. 
No one can open sleeper with sdd dude in it but probably security officers can.
To prevent loosing expensive things there is redemption console near sleepers that is accessible by heads of staff and they can get back thingis that was on sleeping dude that was sucked. 
Redemption console also stores logs of dudes who was sucked in long term storage. 
Yeh and also there was radio announce something like "Greytide_name enters long term storage."

**Comment by Michael-Ashfield** [comment]
For the sake of revs, does this count as a head/rev hiding in a locker or going off the Z-level? It would be annoying to spend 40min looking for the last head only to eventually find them snoozing in a sleeper.

**Comment by AWalton** [comment]
@kopoba BeeStation has that version of cryosleepers as well, but the purpose of these are different from this PR. 

Tau/Bee cryosleepers are for people who know they need to leave the round permanently, so it frees up a job slot for someone else to join the game as. 

This PR cryosleepers are for people who need to go AFK for a short while and intend to come back without suffering theft or harm in the meantime. 

Both versions are very useful, the OP here should consider adding the other type in this PR or a follow-up one.

**Comment by SkeletalElite** [comment]
> @kopoba BeeStation has that version of cryosleepers as well, but the purpose of these are different from this PR.
> 
> Tau/Bee cryosleepers are for people who know they need to leave the round permanently, so it frees up a job slot for someone else to join the game as.
> 
> This PR cryosleepers are for people who need to go AFK for a short while and intend to come back without suffering theft or harm in the meantime.
> 
> Both versions are very useful, the OP here should consider adding the other type in this PR or a follow-up one.

Does everyone commenting not realize you can just use a multitool on it to get someone out? That is  not very safe. Plus on manuel you can just bolt yourself in dorms. For a temporary AFK. It would actually be safer to hide in a locker somewhere.

**Comment by HomocidlePancakes** [comment]
After giving this some though, I realized that this really has no purpose. It’s just a one-tile dorms room. Makes sense to have stuff like that on HRP, because, well ArrPee ‘n shiet, but on TG? Why? Also, I am fairly certain that on bay this would open up the job slot you’ve taken. At least code that it to make this PR somewhat justifiable.

**Comment by Shadowflame909** [comment]
Yeah this is just a locker you multitool instead of busting open or space

**Comment by optimumtact** [comment]
argue about in game terms elsewhere

**Comment by nemvar** [comment]
Why the old sleeper sprites? We have some with open/close animations and directional sprites.

**Review by actioninja — CHANGES_REQUESTED** [review]
use the sleeper sprites with animations

**Comment by Shadowflame909** [comment]
this is codebloat and acts completely like a bolted captains bathroom.

Isn't the function of a cryosleeper to remove someone from the round, reopen their job slot, and keep their stuff secure like the Gulag item storage?

**Comment by Couls** [comment]
I agree with the other comments about this being a remove from round thing rather than "I gotta go pee don't touch my stuff" idea it just seems kind of pointless and there are lockers and stuff to hide in if you gotta go for a bit. though if you can't keep playing on one of the longer rounds its kind of stupid to have a bunch of suicides because of it

File: code/game/objects/structures/crates_lockers/closets/secure/personal.dm (hunk 1 above)
Comment (optimumtact) [reply code/game/objects/structures/crates_lockers/closets/secure/personal.dm]: this needs to go

**Comment by Couls** [comment]
> persist in the doomed world you have created.

**Pull request closed without merging**